balancer.go

```diff
@@ -301,6 +301,12 @@ func (b *balancer) updateConns(newAddrs []resolver.Address, removeConns []conn.C
 			b.pool.RemoveConn(c)
 		}
 	}()
+	var checkClosers []io.Closer
+	defer func() {
+		for _, c := range checkClosers {
+			_ = c.Close()
+		}
+	}()
```

Comment:
This was to prevent risk of deadlock. We don't want to be holding `b.mu` when calling `check.Close()`.